sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -159,14 +159,22 @@ private static ArraySegment<byte> ReadStreamToArraySegment(Stream stream)
                 {
                     using var memStreamCopy = new MemoryStream((int)(memStreamSource.Length - memStreamSource.Position));
```

Comment:
It is actually not necessary to dispose `MemoryStreams` see https://docs.microsoft.com/en-us/dotnet/api/system.io.memorystream?view=netstandard-2.0#remarks but I figured I leave it as is currently since there is no negative impact on the code when we still do it. 

One could argue though that it could be "confusing" to the reader of the code to see that we pass the buffer beyond the scope of the lifetime of the memory stream. Yet not disposing the stream could trigger some kind of analysis warning.

Raising it here as a discussion point to see what you prefer.

Reply:
Personally, I'd keep the `using` to appease the analyzers and keep with the "dispose it if its disposable" heuristic.   Maybe drop a comment that the buffer is safe to outlive the dispose call.